kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/RemotePluginConfigChangedListener.kt

```diff
@@ -0,0 +1,92 @@
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.kork.plugins.remote
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.config.DefaultServiceEndpoint
+import com.netflix.spinnaker.config.okhttp3.OkHttpClientProvider
+import com.netflix.spinnaker.kork.annotations.Beta
+import com.netflix.spinnaker.kork.exceptions.IntegrationException
+import com.netflix.spinnaker.kork.plugins.events.RemotePluginConfigChanged
+import com.netflix.spinnaker.kork.plugins.events.RemotePluginConfigChanged.Status.DISABLED
+import com.netflix.spinnaker.kork.plugins.events.RemotePluginConfigChanged.Status.ENABLED
+import com.netflix.spinnaker.kork.plugins.events.RemotePluginConfigChanged.Status.UPDATED
+import com.netflix.spinnaker.kork.plugins.remote.transport.OkHttpRemoteExtensionTransport
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationListener
+
+@Beta
+class RemotePluginConfigChangedListener(
+  private val objectMapper: ObjectMapper,
+  private val okHttpClientProvider: OkHttpClientProvider,
+  private val remotePluginsCache: RemotePluginsCache
+) : ApplicationListener<RemotePluginConfigChanged> {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun onApplicationEvent(event: RemotePluginConfigChanged) {
+    when (event.status) {
+      ENABLED -> put(event)
+      UPDATED -> put(event)
+      DISABLED -> remotePluginsCache.remove(event.pluginId)
+    }
+  }
+
+  private fun put(event: RemotePluginConfigChanged) {
+    val remoteExtensions: MutableSet<RemoteExtension> = mutableSetOf()
+
+    event.remoteExtensionConfigs.forEach { remoteExtensionConfig ->
+
+      // TODO(jonsie): Support enabling/disabling transports in the config.
+      // Configure HTTP if it is available since it is the only configurable transport right now.
+      val remoteExtensionTransport = if (remoteExtensionConfig.transport.http.url.isNotEmpty()) {
+        val client = okHttpClientProvider.getClient(
+          DefaultServiceEndpoint(
+            remoteExtensionConfig.id,
+            remoteExtensionConfig.transport.http.url,
+            remoteExtensionConfig.transport.http.config
+          )
+        )
+        OkHttpRemoteExtensionTransport(
+          objectMapper,
+          client,
+          remoteExtensionConfig.transport.http.url
+        )
+      } else {
+        throw RemoteExtensionTransportConfigurationException(event.pluginId)
+      }
+
+      remoteExtensions.add(
+        RemoteExtension(
+          remoteExtensionConfig.id,
+          remoteExtensionConfig.type,
+          remoteExtensionConfig.config ?: mapOf(),
+          remoteExtensionTransport
+        )
+      )
+    }
+
+    val remotePlugin = RemotePlugin(event.pluginId, event.version, remoteExtensions)
+    remotePluginsCache.put(remotePlugin)
+    log.debug("Remote plugin '{}' added to cache due to '{}' event", event.pluginId, event.status.toString())
```

Comment:
```suggestion
    log.debug("Remote plugin '{}' added to cache due to '{}' event", event.pluginId, event.status)
```

nit: `toString()` will be called for you.